src/nvim/syntax.c

```
@@ -3286,17 +3288,28 @@ static void syn_cmd_off(exarg_T *eap, int syncing)
 }
 
 static void syn_cmd_onoff(exarg_T *eap, char *name)
+  FUNC_ATTR_NONNULL_ALL
 {
-  char buf[100];
-
+  did_syntax_onoff = true;
   eap->nextcmd = check_nextcmd(eap->arg);
   if (!eap->skip) {
-    strcpy(buf, "so ");
+    char buf[100];
+    strncpy(buf, "so ", 3);
```

Something like that.. but might need some more conditions to avoid hiding errors in the wrong cases.